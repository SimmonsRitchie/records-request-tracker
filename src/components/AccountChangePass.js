import React from 'react';

export class AccountChangePass extends React.Component {
    state = {
        newPass1: "",
        newPass2: "",
        msg: ""
    }
    handleChangePass = () => {
            if (this.state.newPass1 !== this.state.newPass2) {
                this.setState(() => ({
                    msg: "Password fields don't match. Retype new password"
                }))   
            } else {
                this.props.user.updatePassword(this.state.newPass1).then(() => {
                        console.log("Password changed")    
                        this.setState(() => ({
                            msg: "Password changed"
                        }))   
                }).catch((error) => {
                        console.log("Password unchanged")
                        this.setState(() => ({
                            msg: error.message
                        }))
            })
        }
    }

    handleOnChange = (e) => {
        this.setState({ [e.target.name]: e.target.value });
    };

    render() {
        return (
            <div className="account__container">
                <input
                    name="newPass1"
                    type="password"
                    className="text-input"
                    placeholder="Enter new password"
                    onChange={this.handleOnChange}
                    value={this.state.newPass1}
                />
                <input
                    name="newPass2"
                    type="password"
                    className="text-input"
                    placeholder="Confirm new password"
                    onChange={this.handleOnChange}
                    value={this.state.newPass2}
                />
                {this.state.msg && <p className="form__error">{this.state.msg}</p>}
                <button
                    className="button"
                    onClick={this.handleChangePass}
                >
                    Set new password
                </button>
            </div> 
        )
    }
}
