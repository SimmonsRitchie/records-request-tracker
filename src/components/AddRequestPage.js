import React from 'react';
import RequestForm from './RequestForm'

const AddRequestPage = () => (
    <div>
    <p>Add request</p>
    <RequestForm />
    </div>
)

export default AddRequestPage;