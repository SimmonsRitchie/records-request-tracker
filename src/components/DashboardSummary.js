import React from 'react';
import {Link} from 'react-router-dom'


const DashboardSummary = () => (
    <div className="page-header">
        <div className="content-container">
            <h1 className="page-header__title">
                You have filed XX requests
            </h1>
            <h2 className="page-header__title">
                Viewing X because of filters
            </h2>
            <div className="page-header__actions">
                <Link className="button" to="/create">Add request</Link>
            </div>
        </div>
    </div>
)

export default DashboardSummary;