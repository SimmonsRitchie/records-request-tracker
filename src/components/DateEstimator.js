import React from 'react';
import dateEstimator from '../selectors/dateEstimator'

const DateEstimator = ({status, estInterimResponseDate, estFinalResponseDate, estAppealDeadline, estFinalDetermDate}) => (
    <div>
        <p className="date-estimator__title">
        Estimated timeline:
        </p>
        <p className="date-estimator__text">
            {dateEstimator(status, {
                estInterimResponseDate,
                estFinalResponseDate,
                estAppealDeadline,
                estFinalDetermDate,
            })}
        </p>
    </div>
)

export default DateEstimator;