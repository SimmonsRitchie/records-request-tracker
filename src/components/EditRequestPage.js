import React from 'react';
import { connect } from 'react-redux'
import RequestForm from './RequestForm'
import PageHeader from './PageHeader'
import { startEditRequest, startRemoveRequest } from '../actions/requests'
import RemovalModal from './RemovalModal'


/* What's happening here:
1. We pass down two props to RequestForm.
2. Firstly, we pass down the request info as the 'request' props. We get this info from redux.
In mapStateToProps we get data for the request that matches the ID of the 
URL. Eg. /edit/123 will get request data for the request with ID '123' 
3. Then, like AddRequestPage, we pass down the editRequest action dispatcher
to RequestForm under the 'onSubmit' prop.
4. When a user hits the 'save' button in RequestForm, it triggers onSubmit.
5. In this case, because we're feeding the 'editRequest' action instead of the
'addRequest' action, it triggers the 'editRequest' action when a user clicks the 'save' button.

*/

export class EditRequestPage extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            modalIsOpen: false
        };
    };
    startEditRequest = (request) => {
        this.props.startEditRequest(this.props.request.id, request)
        this.props.history.push('/')
    };
    openModal= () => {
        this.setState({modalIsOpen: true});
    }
    closeModal= () => {
        this.setState({modalIsOpen: false});
    }
    onRemove = (id) => {
        this.setState({modalIsOpen: false});
        this.props.startRemoveRequest(this.props.request.id);
        this.props.history.push('/')
    };
    render() { 
        return (
        <div>
            <PageHeader 
                pageTitle={"View/Edit request"}
            />
            <div className="content-container">
                <RequestForm 
                    onSubmit={this.startEditRequest}
                    request={this.props.request}
                />
            <button
                className="button button--remove"
                onClick={this.openModal}
            >
                Remove request
            </button>
            </div>
            <RemovalModal
                modalIsOpen={this.state.modalIsOpen}
                onRemove={this.onRemove}
                closeModal={this.closeModal}
            />
        </div>
        )
    }
}


// Request form is populated with the expense of the ID that matches
// the ID in URL (eg. /edit/123)
const mapStateToProps = (state, props) => {
    return {
        request: state.requests.find((request) => request.id === props.match.params.id)
    };
};

const mapDispatchToProps = (dispatch) => ({
    startEditRequest: (id, request) => dispatch(startEditRequest(id, request)),
    startRemoveRequest: (id) => dispatch(startRemoveRequest(id))
})

export default connect(mapStateToProps, mapDispatchToProps)(EditRequestPage);