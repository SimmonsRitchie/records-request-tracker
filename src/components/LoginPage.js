import React from 'react';
import { connect } from 'react-redux';
import { startGoogleLogin } from '../actions/auth'
import { CSSTransition, TransitionGroup } from 'react-transition-group'
import LoginPageInitial from './LoginPageInitial'
import LoginPageEmailSignUp from './LoginPageEmailSignUp'
import LoginPageEmailSignIn from './LoginPageEmailSignIn'



class LoginPage extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            display: "loginInitial",
            appearHome: true
        }
    };

    handleEmailLogin = () => {
        this.setState(() => ({
            display: "emailSignIn"
        }))     
    };

    handleGoogleLogin = () => {
        console.log("Logging in with google!")
        this.props.startGoogleLogin()
    };

    handleEmailSignUp= () => {
        this.setState(() => ({
            display: "emailSignUp"
        }))     
    };

    handleBackToStart = () => {
        this.setState(() => ({
            display: "loginInitial",
        }))     
    }



    displayBox = () => {
        if (this.state.display === "loginInitial") {
            return <LoginPageInitial
                handleEmailLogin={this.handleEmailLogin}
                handleGoogleLogin={this.handleGoogleLogin}
                handleEmailSignUp={this.handleEmailSignUp}
                />
        } else if (this.state.display === "emailSignUp") {
            return <LoginPageEmailSignUp
                handleBackToStart={this.handleBackToStart}
                />
        } else if (this.state.display === "emailSignIn") {
            return <LoginPageEmailSignIn
                handleBackToStart={this.handleBackToStart}
                />
        }
    }

    render() {
        return (
            <div className="box-layout">
            <div className="box-layout__box">
                <h1 className="box-layout__title">Right-To-Know Tracker</h1>

                    <TransitionGroup
                        key={this.state.display}
                    >
                        <CSSTransition
                            in={this.state.appearHome}
                            appear={true}
                            classNames="fade"
                            timeout={300}
                        >
                            {this.displayBox}
                        </CSSTransition>
                    </TransitionGroup>
            </div>
        </div>
        )
    };
}

const mapDispatchToProps = (dispatch) => ({
    startGoogleLogin: () => dispatch(startGoogleLogin())
})

export default connect(undefined, mapDispatchToProps)(LoginPage)


