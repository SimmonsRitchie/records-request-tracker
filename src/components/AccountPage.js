import React from 'react';
import {firebase} from '../firebase/firebase'
import PageHeader from './PageHeader'
import {AccountChangePass} from './AccountChangePass'


class AccountPage extends React.Component {
    state = {
        changePass: false
    }
    handleChangePass = () => {
        this.setState(() => ({
            changePass: !this.state.changePass
        }))   
    }

    render() {
        const user = firebase.auth().currentUser

        return (
            <div>
                <PageHeader 
                    pageTitle={"Account settings"}
                />
                <div className="content-container">
                    <h3>Email address: </h3>
                    <p>{user.email}</p>
                    <h3>Display name:</h3>
                    <p>{user.displayName ? user.displayName : "No display name set"}</p>
                    <button
                        className="button"
                        onClick={this.handleChangePass}
                    >
                        Change password
                    </button>
                    {this.state.changePass && <AccountChangePass user={user} />}
                </div>
            </div> 
        )
    }
}


export default AccountPage
