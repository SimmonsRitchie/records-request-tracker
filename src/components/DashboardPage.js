import React from 'react';
import RequestsListFilters from './RequestsListFilters'
import RequestsList from './RequestsList'
import DashboardSummary from './DashboardSummary';


const DashboardPage = () => (
    <div>
        <DashboardSummary />
        <RequestsListFilters />
        <RequestsList />
    </div>
);

export default DashboardPage;