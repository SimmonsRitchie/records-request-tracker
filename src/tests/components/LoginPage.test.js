import React from 'react';
import { LoginPage } from '../../components/LoginPage'
import { shallow } from 'enzyme'

// Using beforeEach, we can save ourselves from re-writing spies
// and the wrapper.
let startLogin, wrapper;

beforeEach(() => {
    startLogin = jest.fn();
    wrapper = shallow(<LoginPage 
        startLogin={startLogin}
        />)
})


test('Should render LoginPage', () => {
    expect(wrapper).toMatchSnapshot()
})

test('Should call startLogin on button click', () => {
    // Below, we find the button tag and trigger 'onClick'
    // by simulating a click.
    wrapper.find('button').simulate('click')
    // We then check to see whether startLogin was
    // called
    expect(startLogin).toHaveBeenCalled()
});