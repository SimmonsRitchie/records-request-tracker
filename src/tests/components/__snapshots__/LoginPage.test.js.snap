// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render LoginPage 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      Right-To-Know Tracker
    </h1>
    <p>
      Organize your open records requests in one handy location.
    </p>
    <button
      className="button"
      onClick={[Function]}
    >
      Login with Google
    </button>
  </div>
</div>
`;
